### DISCUSSION TITLE
InterestPolicy variation

### DISCUSSION THREAD

**Discussion by agufagit** [nats-io/nats-server#4218] [Q&A]
Is there a way not to delete messages immediately if there are no consumers in JetStream?

My use case is that there can be 0 or 1 or multiple consumers for the same subject, and messages published can't be deleted right away if there are no consumers for that subject.  Once messages are consumed at least once, they can be deleted.

I tried default LimitsPolicy to manually delete message by sequence number, if there are multiple consumers, there will be multiple delete requests.  But it causes race condition due to network latency, if message is not deleted fast enough, new consumer will consume that message, which is not desired.

Anyway to make this happen?

Thank you

**Comment by derekcollison** [comment]
When you have more then one consumer do they all need to process all messages or do you mean multiple subscribers where any given subscriber receives a message but only one of them does?

**Reply by agufagit** [reply]
All of consumers will process messages.  Scenario is that user can login to same account from multiple devices (phone, tablet, pc, etc.), and messages should be delivered to all the devices that are logged in.

**Reply by derekcollison** [reply]
Could there be a consumer that is setup when the user registers and is the designated main device and it never goes away?

**Reply by agufagit** [reply]
No, when user logged off, there will be no consumer.  The architecture I'm going with is that, each user has its own subject, so a stream could have millions of subjects.  The messages for that user will be pushed to that subject, and will wait until user log in.

**Reply by derekcollison** [reply]
The consumer can be inactive, it will just be present to make sure messages are not deleted.

**Reply by agufagit** [reply]
After message is consumed at least once, will message be deleted with an inactive consumer?  I can't make that inactive consumer to become active because user can init websocket connections with different load balancing servers

**Reply by derekcollison** [reply]
You will have persistent consumer state for each device, but those consumers are only active when the device is online and connected. Will work fine for you.
